### ISSUE TITLE
quick allow all scripts, but block ads (like NoScript off / ABP on)

### ISSUE THREAD

**Issue by Cimbass** [gorhill/uBlock#1144]
I have set up default restrict blocking all scripts in global dynamic rules, but sometimes, i want allow all scripts temporarily , but however still block ads (like Noscipt off and ADP on)

Current, i must click 4x in dynamic rules to allow this, therefore, a quick option to switch scripts on where nice.

The power button or deactivate the AddOn loads also ads, which i don't want.

All Resources local set does not work (see screenshot)
![zwischenablage01](https://cloud.githubusercontent.com/assets/16479673/12039952/147b32e8-ae65-11e5-8e5e-fba69778db70.jpg)

**Comment by RoxKilly** [comment]
@Cimbass This is not exactly what you want but you could open uBO settings and uncheck "I am an advanced user".  This will disable all your dynamic filters but keep your static filters.  In effect it will allow all scripts except those blocked by the 3rd party filters you have turned on. 

Once you re-check that settings, your dynamic filters will be restored as they were. See the [wiki](https://github.com/gorhill/uBlock/wiki/Advanced-user-features)

It's not a 1-click solution, and affects all sites rather than a targeted site, but it's one option if you want to just temporarily put down your shields.

**Comment by Cimbass** [comment]
Thanks for your help, that would do to me, but this works for some reasons not really correctly!

- Go with advanced user check (and my settings in the screenshot) to http://www.test-ipv6.com/ and it will show the "JavaScript erforderlich" message
- uncheck advanced user and reload the page shows sadly again the Javascript warning

- but then, when you switch the big power button off (gray) and reload, than the page works fine (but loads also ads - not in this site, but in others = bad)
- switch on the power button (blue) and reload, page than again works fine

And only switch the power button off (without reloading) and on and than reload does also not work.

**Comment by gorhill** [comment]
> sometimes, i want allow all scripts temporarily

If it's only "sometimes", why is it an issue? Thing is I am very conservative with any request to change the UI, and the one here is such a request. Adding yet another row for the sake in the dynamic filtering panel does not strike me as a good idea: this push down further the list of domain names, this is redundant to what is already there, this would require mode docs to explain what it is about, and so on.

So for now I will decline. As usual, I may or may not change my mind in some future.

**Comment by Cimbass** [comment]
OK, but have you read my second post? There seems for me to be a bug in switching advanced settings on/off while remained on the actual page? When this will works like RoxKilly described, no change in the UI is necessary and I am happy ;)

Or should I open a new issue for that?

**Comment by gorhill** [comment]
> And only switch the power button off (without reloading) and on and than reload does also not work.

This is probably the same bug reported in uMatrix: <https://github.com/gorhill/uMatrix/issues/396#issuecomment-151573246>.

Summary: click uBO's reload button should work fine, click the browser's reload button may not work. Shift-click the browser's reload button should work.

**Comment by Cimbass** [comment]
OK, thanks, shift-click works.

Then, what about this suggestion: when changing the advanced user setting, then show in uBlock the reload button?

**Comment by gorhill** [comment]
Sometimes trying to cram in doubtful workarounds for uncommon issues affecting a minority _sometimes_ lead to overcomplicating the software for everybody. <kbd>Shift</kbd>-click is a good enough workaround for your case, without overcomplicating uBO.
